dbms/src/Flash/Mpp/AsyncRequestHandler.h

```diff
@@ -394,5 +397,8 @@ class AsyncRequestHandler : public AsyncRequestHandlerBase
 
     // try-catch may call close_conn, and we need to ensure that close_conn is called for only once.
     bool is_close_conn_called;
+
+    std::mutex mu;
```

Comment:
It looks like `std::mutex mu` and `std::mutex make_reader_mu` can be merged into one

Reply:
> It looks like `std::mutex mu` and `std::mutex make_reader_mu` can be merged into one

No, or it will be dead lock as we will call `retryOrDone` in `processWaitMakeReader`. I think we can rename mu to `is_close_conn_called_mu`.

Reply:
ok